src/utils/event.ts

```diff
@@ -181,15 +181,33 @@ export const identifyEvent = async (event: UnidentifiedEvent): Promise<UnsignedE
   return { ...event, id }
 }
 
-export function getRelayPrivateKey(relayUrl: string): string {
+let privateKeyCache: string | undefined
+export function getRelayPrivateKey(secret: string): string {
+  if (privateKeyCache) {
+    return privateKeyCache
+  }
+
   if (process.env.RELAY_PRIVATE_KEY) {
-    return process.env.RELAY_PRIVATE_KEY
+    privateKeyCache = process.env.RELAY_PRIVATE_KEY
+
+    return privateKeyCache
   }
 
-  return deriveFromSecret(relayUrl).toString('hex')
+  privateKeyCache = deriveFromSecret(secret).toString('hex')
+
+  return privateKeyCache
 }
 
-export const getPublicKey = (privkey: string | Buffer) => Buffer.from(secp256k1.getPublicKey(privkey, true)).subarray(1).toString('hex')
+const publicKeyCache: Record<string, string> = {}
+export const getPublicKey = (privkey: string) => {
+  if (privkey in publicKeyCache) {
+    return publicKeyCache[privkey]
+  }
+
+  publicKeyCache[privkey] = Buffer.from(secp256k1.getPublicKey(privkey, true)).subarray(1).toString('hex')
```

Comment:
You can do something like this instead:
`secp256k1.utils.bytesToHex(secp256k1.getPublicKey(privkey, true)).subarray(1))`